### PR TITLE
Write opcodes to file only in debug mode

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose
Write opcodes to file only in debug mode.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 1]
```diff
@@ -139,7 +148,14 @@ internal CodeGenIL(IDictionary<string, IList> input, string filePath, ProtoCore.
                 }
                 EmitOpCode(OpCodes.Ret);
                 return (asm, type);
+#if DEBUG
             }
+#endif
+            EmitOpCode(OpCodes.Ret);
+#if DEBUG
+            writer.Close();
```
Comment (mjkkirschner) [inline src/Engine/EmitMSIL/CodeGenIL.cs:156]: why do we need to call writer.Close() if the writer is wrapped in a using?

Comment (aparajit-pratap) [reply src/Engine/EmitMSIL/CodeGenIL.cs:156]: I think this line reappeared after I resolved merge conflicts with my local branch, will remove it.

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 2]
```diff
@@ -848,50 +864,64 @@ public Type DfsTraverse(Node n)
...
+#if DEBUG
             writer.WriteLine($"{opCode} {index}");
+#endif
         }
 
         private void EmitOpCode(OpCode opCode, string str)
         {
             if (compilePass == CompilePass.GatherTypeInfo) return;
             ilGen.Emit(opCode, str);
+#if DEBUG
```
Comment (mjkkirschner) [inline src/Engine/EmitMSIL/CodeGenIL.cs:922]: I guess ideally we would wrap this 3 lines into a function? maybe a tiny local one?

**Comment by aparajit-pratap** [comment]
Merging immediately so I can trigger the perf CI again on this branch.

**Pull request merged**